agent/accumulator.go

```diff
@@ -161,6 +164,14 @@ func (ac *accumulator) AddFields(
 	ac.metrics <- m
 }
 
+func (ac *accumulator) AddError(err error) {
+	if err == nil {
+		return
+	}
+	atomic.AddUint64(&ac.errCount, 1)
```

Comment:
to illustrate the point, atomic.Add... doesn't directly increment the counter, it returns a new counter that has been incremented: https://play.golang.org/p/7y7RAxUjJs